### ISSUE TITLE
Grid: Value in aria-sort is not perceptible

### ISSUE THREAD

**Issue by JAWS-test** [w3c/aria-practices#1175]
https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/dataGrids.html: "Example 2: Sortable Data Grid With Editable Cells"

The value in the aria-sort attribute is not output by JAWS 2019 (with current versions of IE 11, Chrome, and Firefox) when you navigate the table with arrow keys in forms mode. I think that is not a bug of JAWS or of the browser, but a problem of design. The column header with aria-sort does not get the focus, but the buttons inside it. However, assistive technology cannot be expected to output ARIA attributes of ancestor elements. 

I don't know what the best solution for the problem would be, because it makes sense to have the buttons get the focus instead of the column headers.

**Comment by ZoeBijl** [comment]
> However, assistive technology cannot be expected to output ARIA attributes of ancestor elements.

Why do you think that?

---

Perhaps the linkage between the `button` and the `th` should be made clear by an `aria-controls`. The `button` does control the column’s ordering but it doesn’t directly control the `th`. So this is a very questionable suggestion.

In the end the information is available in the DOM/accessibility tree and not communicating it to users is a failure of the assistive technology.

**Comment by JAWS-test** [comment]
@ZoeBijl 
> Why do you think that?

I assumed that an ARIA attribute always applies only to the current element and not to ancestors or descendants. It also seems wrong to me if all ARIA attributes would be output automatically for all descendant elements. This would lead to a wrong and too detailed output. As far as I can see, attributes that affect descendant elements are explicitly specified in the ARIA specification, such as aria-hidden and aria-disabled. If my assumption is wrong, perhaps the ARIA specification should contain a hint whether all attributes should always be output at each descendant element.

**Comment by ZoeBijl** [comment]
This is an AT bug. I’m locking this thread because it’s not productive. If you have further comments you can open a new issue on the condition that you provide the following:

- A working live test
- Links to the various bits in the ARIA specification you think are wrong
